### ISSUE TITLE
Zoom panes independently

### ISSUE THREAD

**Issue by hyiltiz** [wez/wezterm#3116]
**Is your feature request related to a problem? Please describe.**
I keep a few panes open and zoom out from some of the panes where I just keep monitoring programs (like top, watch, ping etc.) but work in a large pane (e.g. vi).

**Describe the solution you'd like**
Add ability to zoom each pane independently.

**Describe alternatives you've considered**
iTerm2

**Additional context**
`wezterm 20221119-145034-49b9839f`

**Comment by wez** [comment]
I don't understand this request; can you be more specific about what you want to see?

**Comment by snaggen** [comment]
Isn't that what the keys
```
        CTRL                 Z                  ->   TogglePaneZoomState
        SHIFT | CTRL         Z                  ->   TogglePaneZoomState
        SHIFT | CTRL         z                  ->   TogglePaneZoomState
``` 
does? or....

**Comment by hyiltiz** [comment]
No, TogglePaneZoomState seems to effectively hide the rest of the frames in a given tab/window, not change the font size of a certain frame.
<img width="1305" alt="image" src="https://user-images.githubusercontent.com/3246683/219341670-793bb84c-67b3-4562-930b-53a3f9a384d7.png">

**Comment by putrasattvika** [comment]
I'm not really sure how iTerm2 behaves, but currently I'm using [Terminator](https://github.com/gnome-terminator/terminator) which has the feature [hyiltiz](https://github.com/hyiltiz) described: font size adjustment (<kbd>Ctrl</kbd> + <kbd>-</kbd> and <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>+</kbd>) will only change the font size of the currently active pane.

My personal use case is that sometimes I need to work with commands that produce wide output (e.g. `kubectl get pods --all-namespaces -o wide`) and I need to reduce the pane's font size so that long lines won't get wrapped, while at the same time I want all other panels to remain unchanged.

**Comment by rgilio** [comment]
I'm in the same situation. Migrating from terminator over to wezterm but having issues with the font sizing being global and not per pane. Changing the font size also affects the other tabs which is a bit unexpected. 

If there is now a feature that allows per-pane font sizing, I'm not aware of it and can't find anything in the current documentation.

Are there any plans to add per-pane font sizing?

**Comment by ww7** [comment]
Also wondering for this feature.

**Comment by krskrs** [comment]
Has anything changed regarding setting the font_size independently per pane? I couldn't find anything in the documentation and ended up here after googling for it.

**Comment by seflue** [comment]
I also actually ended up here while googling for this feature. Ghostty actually has it but I was wondering, if it is technically possible to implement it in Wezterm.
Especially when using the terminal multiplexing capabilities, it could be really useful to be able to adjust the font size independently for each pane.

**Comment by seflue** [comment]
> > Ghostty actually has it but I was wondering, if it is technically possible to implement it in Wezterm.
> 
> Please avoid unnecessary comments like this, comparison are no good, this is a small project and thing like this takes time.

It was not my intention to force any comparison race. I only mentioned Ghostty, because of @wez [comment](https://github.com/wez/wezterm/issues/3116#issuecomment-1432103507). Being able to try it out in another project is sometimes clearer than any explanation.

It was not my intention to make an unnecessary comment, I would just like to know if wezterms architecture would allow to add such a feature without big hurdles or if it would be more complicated. And if it would fit the vision of the creators and has a chance to be added or not. I don't want to hurt anyone's feelings here. I really appreciate this project and all the time, work and passion that went into it.

**Comment by hyiltiz** [comment]
For anyone who is familiar with wezterm's internals, any pointers for what the ideal implementation would look like, by filling out the form below?

- Files directly relevant (exposes functionality relevant to panel creation, pane state modification, text rendering, key binding): 
- Files likely expected to be modified: 
- Tests that might be relevant: 

So someone new might be able to pick up the task?

**Comment by dominiwe** [comment]
@hyiltiz I really want this feature too and will try looking into it next week. If you already did some of the research you mentioned, let me know. Will link my branch once I have something working to show. I think it would be best to start by putting this behind a feature flag regardless of everything that may need to be changed (could be quite a lot).

**Comment by evs-xsarus** [comment]
@dominiwe maybe add a new keybinding; thus keep the current one for global increase/decrease font size and use 

CTRL+SHIFT+-/= for a pane specific zoom

A feature flag can then be skipped.
